crates/red_knot_server/src/server/api/notifications/did_close.rs

```diff
@@ -34,10 +32,6 @@ impl SyncNotificationHandler for DidCloseTextDocumentHandler {
             .close_document(&key)
             .with_failure_code(ErrorCode::InternalError)?;
 
-        if let Some(db) = session.workspace_db_for_path_mut(path.as_std_path()) {
-            File::sync_path(db, &path);
-        }
-
```

Comment:
I don't think it's required for the server to sync the file when it's closed. The concern here was that when an unsaved file content is either accepted or discarded, it might require us to sync it but the client will send us the `didChange` request before the `didClose`. So, at this point the file content is already in sync.

Reply:
Shouldn't that need to happen explicitly by updating the file status? This means it requires a dedicated `ChangedEvent` as done in https://github.com/astral-sh/ruff/pull/13044.